app/Models/Foundation/Summit/Registration/ExtraQuestions/SummitOrderExtraQuestionType.php

```diff
@@ -102,12 +125,47 @@ public function getExternalId(): ?string
     }
 
     /**
-     * @param string $external_id
+     * @param string|null $external_id
      */
     public function setExternalId(?string $external_id): void
     {
         $this->external_id = $external_id;
     }
 
+    /**
+     * @return SummitTicketType[]
+     */
+    public function getAllowedTicketTypes(): array
+    {
+        return $this->allowed_ticket_types->toArray();
+    }
+
+    /**
+     * @param SummitTicketType $ticket_type
+     */
+    public function addAllowedTicketType(SummitTicketType $ticket_type): void
+    {
+        if ($this->allowed_ticket_types->contains($ticket_type)) return;
+        $this->allowed_ticket_types->add($ticket_type);
+    }
+
+    /**
+     * @return SummitBadgeFeatureType[]
+     */
+    public function getAllowedBadgeFeatureTypes(): array
+    {
+        return $this->allowed_badge_features_types->toArray();
```

Comment:
this is not very performant way bc its allocates the entire collection and u lost the lazy load collection feature

Reply:
done